omegaconf/basecontainer.py

```diff
@@ -237,13 +240,16 @@ def str_representer(dumper: yaml.Dumper, data: str) -> yaml.ScalarNode:
                 style=("'" if with_quotes else None),
             )
 
-        dumper.add_representer(str, str_representer)
-        container = OmegaConf.to_container(self, resolve=resolve, enum_to_str=True)
+        Dumper.add_representer(str, str_representer)
+        container = OmegaConf.to_container(
+            self, resolve=kwds["resolve"], enum_to_str=True
```

Comment:
so _dump is just making the rest of the code more complicated.
let's get rid of it, I didn't mean this.